### ISSUE TITLE
todolist directive generates wrong hyperlinks in LaTeX PDF output

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#12520]
### Describe the bug

The LaTeX mark-up  for ``todo`` directives uses a `\label` which can not work because there is no associated counter.  As a result from the Todo List, the links go generally to the section title, not to the actual todo item.

Here is a screenshot where the target of the second item in the todo list is shown.
![Capture d’écran 2024-07-08 à 22 09 46](https://github.com/sphinx-doc/sphinx/assets/2589111/814bacc9-be1e-4979-a4c8-1a165c051916)

### How to Reproduce

``index.rst``

```rest
Welcome to FOO's documentation!
===============================

TEST
====

section 1
---------

.. todo::

   Implement TODOs!


Some text


Some more text

.. todo::

   Implement TODOs! Do it ! (2)


section 2
---------

.. todolist::
```

``conf.py``

```
project = 'FOO'
copyright = '2024, JFB'
author = 'JFB'
version = '0'
release = '0'
extensions = ['sphinx.ext.todo']
todo_include_todos = True
todo_link_only = True
language = 'en'
# master_doc = 'index'
```

### Environment Information

```text
Sphinx version: 7.4.0+/7eb77f237
Python version: 3.12.3 (CPython)
Docutils version: 0.21.2 
Pygments version: 2.18.0
Jinja2 version: 3.1.4
```

### Sphinx extensions

```python
['sphinx.ext.todo']
```

**Comment by jfbu** [comment]
This was fixed by merge of #12508. An update to CHANGES.rst will be done a posteriori later. Closing.

**Comment by jfbu** [comment]
This was actually a duplicate of #8807.
